src/vmm/src/arch/x86_64/mod.rs

```diff
@@ -456,6 +472,56 @@ pub fn load_kernel(
     })
 }
 
+#[cfg(kani)]
+mod verification {
+    use crate::arch::x86_64::FIRST_ADDR_PAST_32BITS;
+    use crate::arch::{MMIO_MEM_START, arch_memory_regions};
+
+    #[kani::proof]
+    #[kani::unwind(3)]
+    fn verify_arch_memory_regions() {
+        let offset: u64 = kani::any::<u64>();
+        let len: u64 = kani::any::<u64>();
+
+        kani::assume(len > 0);
+        kani::assume(offset.checked_add(len).is_some());
+
+        let regions = arch_memory_regions(offset as usize, len as usize);
+
+        // There's only one MMIO gap, so we can get either 1 or 2 regions
+        assert!(regions.len() <= 2);
```

Comment:
mainly a note to myself: PCIe might change this

Reply:
Just out of curiosity, how so? :o

Reply:
Ah, that makes sense. Will probably end up having to take that into account when it comes to allocating the swiotlb region over in #5108, since I guess we'll get PCI before secret hiding, but shouldn't be a problem tbh.